### PR TITLE
Jazzbar responsive layout for 4+ people

### PR DESCRIPTION
**Pull request by mike-lvov**
Jazzbar responsive layout
closes sparkletown/internal-sparkle-issues#317
closes https://github.com/sparkletown/internal-sparkle-issues/issues/467
closes https://github.com/sparkletown/internal-sparkle-issues/issues/466
closes https://github.com/sparkletown/internal-sparkle-issues/issues/464

### CODE REVIEW

**Comment by 0xdevalias** [comment]
@mike-lvov If this is not for review, probably best for it to be a draft PR?

**Comment by mike-lvov** [comment]
> @mike-lvov If this is not for review, probably best for it to be a draft PR?

I thought it would not create a Preview, if I made it as draft, but it doesn't seem to be the case

**Comment by 0xdevalias** [comment]
> I thought it would not create a Preview, if I made it as draft, but it doesn't seem to be the case

@mike-lvov Yeah, that's fair. It should run for all types of PRs (keeping in mind the "doesn't seem to update" issue we discussed on slack, which is still an issue at the moment I believe)

**Comment by mike-lvov** [comment]
I don't feel good about the way I implemented handling different screen sizes, so I'm open to suggestions about how to approach it in a neater way

**Comment by 0xdevalias** [comment]
> I don't feel good about the way I implemented handling different screen sizes, so I'm open to suggestions about how to approach it in a neater way

@mike-lvov Reminder to make sure to request the review when you want it reviewed. That's probably why it didn't show up on my list of things to look at earlier today.

File: src/components/templates/Jazzbar/JazzTab/JazzTab.scss
Comment (0xdevalias) [inline src/components/templates/Jazzbar/JazzTab/JazzTab.scss:41]: These styles appear to be used in `src/components/templates/Jazzbar/components/JazzBarTableComponent/JazzBarTableComponent.tsx`, do they need to be in this file?

If they're custom overrides I wonder if we can have them somewhere clearer feelings/not necessarily as strictly nested as they are here. Would they still work if they were in the 'root' of the block, rather than nested under `music-bar-content`?

File: src/components/templates/Jazzbar/JazzTab/JazzTab.scss
Comment (0xdevalias) [reply src/components/templates/Jazzbar/JazzTab/JazzTab.scss:9]: Basically my thinking here is that if can 'unnest' this, then it will probably be easier to put all of the mediaqueries 'together', which I think would make this feel a lot cleaner/simpler to follow.

File: src/components/templates/Jazzbar/JazzTab/JazzTab.scss
Comment (0xdevalias) [inline src/components/templates/Jazzbar/JazzTab/JazzTab.scss:16]: What is this `items` section used for? I didn't see any `grid-area` referencing it?

File: src/components/templates/Jazzbar/JazzTab/JazzTab.scss
Comment (0xdevalias) [inline src/components/templates/Jazzbar/JazzTab/JazzTab.scss:80]: Do we still account for the 16:9 needs here?

File: src/components/templates/Jazzbar/JazzTab/JazzTab.scss
Comment (0xdevalias) [inline src/components/templates/Jazzbar/JazzTab/JazzTab.scss:93]: Where does this `90%` come from?

**Review by 0xdevalias — CHANGES_REQUESTED** [review]
A whole bunch of comments/suggestions. Not sure if they're going to be helpful or just cause more confusion.. but hopefully they'll be helpful!

File: src/components/templates/Jazzbar/JazzTab/JazzTab.tsx
Comment (mike-lvov) [reply src/components/templates/Jazzbar/JazzTab/JazzTab.tsx:3]: After it

File: src/components/templates/Jazzbar/JazzTab/JazzTab.tsx
Comment (mike-lvov) [reply src/components/templates/Jazzbar/JazzTab/JazzTab.tsx:190]: Agreed on suggested. I have a similar urge towards the Header / Room. I created this PR as a polish for product guys with the view of further rebuilding/refactor of the component. I expect this change to be in that refactor ^

File: src/components/templates/Jazzbar/components/JazzBarRoom/Room.scss
Comment (mike-lvov) [reply src/components/templates/Jazzbar/components/JazzBarRoom/Room.scss:81]: The reason for it to be nested is specificity. I need this rule to be applied only when the parent is `participants`.

File: src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx
Comment (mike-lvov) [reply src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx:300]: Agreed for [OOS]. Currently, can't do that because it will affect the grid system

File: src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx
Comment (mike-lvov) [reply src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx:230]: The logic is purely for having nice alignment for the bottom row of videos. Grid doesn't allow that, so we have grid for sided videos and flex for otherVideos

File: src/components/templates/Jazzbar/JazzTab/JazzTab.scss
Comment (mike-lvov) [reply src/components/templates/Jazzbar/JazzTab/JazzTab.scss:91]: No, they are unrelated. This reduction was done to get as much space as possible

File: src/components/templates/Jazzbar/JazzTab/JazzTab.scss
Comment (mike-lvov) [reply src/components/templates/Jazzbar/JazzTab/JazzTab.scss]: I don't like the way Header was built in the first place. I don't think we should have such component. The way I see it is that all alignment should happen on Venues side, when we would have only separate components like TableLock/TableInfo/similar. That would be part of my [OOS] for changing JazzBar. This PR is intended to bring external beauty for the coming events

File: src/components/templates/Jazzbar/JazzTab/JazzTab.scss
Comment (mike-lvov) [reply src/components/templates/Jazzbar/JazzTab/JazzTab.scss]: The reason for this margin to be is the `Back to venue` button, which is positioned absolutely. We don't have it in Table view

File: src/components/templates/Jazzbar/JazzTab/JazzTab.tsx
Comment (0xdevalias) [reply src/components/templates/Jazzbar/JazzTab/JazzTab.tsx:3]: Cool, makes sense.

File: src/components/templates/Jazzbar/JazzTab/JazzTab.scss
Comment (mike-lvov) [reply src/components/templates/Jazzbar/JazzTab/JazzTab.scss]: It is in Jazzbar's Room.scss

image

File: src/components/templates/Jazzbar/JazzTab/JazzTab.scss
Comment (mike-lvov) [reply src/components/templates/Jazzbar/JazzTab/JazzTab.scss:83]: Better visual interception. It was done to get as much space for videos as possible. By making `actions container` larger, we took the unused space under the video

File: src/components/templates/Jazzbar/JazzTab/JazzTab.scss
Comment (mike-lvov) [reply src/components/templates/Jazzbar/JazzTab/JazzTab.scss:70]: The video sizes has changed and with new hardcoded video sizes(not the best way, but the most optimal one), we don't need to hardcode the video size

File: src/components/templates/Jazzbar/JazzTab/JazzTab.scss
Comment (mike-lvov) [reply src/components/templates/Jazzbar/JazzTab/JazzTab.scss:31]: Moved them

File: src/components/templates/Jazzbar/components/JazzBarRoom/Room.scss
Comment (mike-lvov) [reply src/components/templates/Jazzbar/components/JazzBarRoom/Room.scss:76]: Picked one and moved it into constants

File: src/components/templates/Jazzbar/components/JazzBarRoom/Room.scss
Comment (mike-lvov) [reply src/components/templates/Jazzbar/components/JazzBarRoom/Room.scss:13]: Given the timing/testing needs, I'd rather merge this first.

File: src/components/templates/Jazzbar/JazzTab/JazzTab.scss
Comment (mike-lvov) [reply src/components/templates/Jazzbar/JazzTab/JazzTab.scss]: Moved to constants

File: src/components/templates/Jazzbar/components/JazzBarRoom/Room.scss
Comment (mike-lvov) [reply src/components/templates/Jazzbar/components/JazzBarRoom/Room.scss:10]: moved to local consts

File: src/components/templates/Jazzbar/JazzTab/JazzTab.scss
Comment (mike-lvov) [reply src/components/templates/Jazzbar/JazzTab/JazzTab.scss]: The reason it is nested is because it should only apply, when there is tableView status. I do agree it looks a bit cluttered, but ` it will probably be easier to put all of the mediaqueries 'together',` we don't really have much of them + this is considered to be a `temporary/quick` polish. I expect the whole component to be rewritten in the nearest future(Probably after the Auditorium)

File: src/components/templates/Jazzbar/JazzTab/JazzTab.tsx
Comment (0xdevalias) [reply src/components/templates/Jazzbar/JazzTab/JazzTab.tsx:190]: Sounds good to me.

File: src/components/templates/Jazzbar/components/JazzBarRoom/Room.scss
Comment (0xdevalias) [reply src/components/templates/Jazzbar/components/JazzBarRoom/Room.scss]: Why does it need to be applied only when that specific parent though? What specificity/scoping requires that here, that the BEM block wouldn't also provide?

File: src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx
Comment (0xdevalias) [reply src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx:300]: _nods_ makes sense

File: src/components/templates/Jazzbar/JazzTab/JazzTab.scss
Comment (0xdevalias) [reply src/components/templates/Jazzbar/JazzTab/JazzTab.scss:76]: Fair enough, makes sense.

File: src/components/templates/Jazzbar/JazzTab/JazzTab.scss
Comment (0xdevalias) [reply src/components/templates/Jazzbar/JazzTab/JazzTab.scss:55]: When you say 'with new hardcoded video sizes', where is this done? Within the media queries, or?

File: src/components/templates/Jazzbar/JazzTab/JazzTab.scss
Comment (mike-lvov) [reply src/components/templates/Jazzbar/JazzTab/JazzTab.scss]: fixed in https://github.com/sparkletown/sparkle/pull/1217/commits/40d0ae8e83b526e9d6395005d992d434148f3c52

File: src/components/templates/Jazzbar/JazzTab/JazzTab.scss
Comment (mike-lvov) [reply src/components/templates/Jazzbar/JazzTab/JazzTab.scss:55]: In local constants. Basically, we have video height as 2 * video participant height (195px) + paddings. It's hard to keep the best 16:9 aspect ratio for every screen size, but it's good enough for all usecases now. This view was approved by @Sparkle-T, @sparkleed

File: src/components/templates/Jazzbar/components/JazzBarRoom/Room.scss
Comment (mike-lvov) [reply src/components/templates/Jazzbar/components/JazzBarRoom/Room.scss]: You have two participant types. The sided ones and those, under the `participants` div. They slightly differ in their styling

image

File: src/components/templates/Jazzbar/components/JazzBarRoom/Room.scss
Comment (0xdevalias) [reply src/components/templates/Jazzbar/components/JazzBarRoom/Room.scss:14]: Yup, i'm fine with that. Was more a 'if it makes your life here easier'.

File: src/components/templates/Jazzbar/JazzTab/JazzTab.scss
Comment (0xdevalias) [reply src/components/templates/Jazzbar/JazzTab/JazzTab.scss:55]: Cool cool, makes sense.

File: src/components/templates/Jazzbar/components/JazzBarRoom/Room.scss
Comment (0xdevalias) [reply src/components/templates/Jazzbar/components/JazzBarRoom/Room.scss]: Yeah ok, that makes sense. Must have missed that nuance when I was reading the code.

File: src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx
Comment (0xdevalias) [reply src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx:230]: While I haven't checked this specifically, this issue I stumbled across suggests that it could be possible without this:

- https://stackoverflow.com/questions/43129360/css-grid-wrapping/43129507#43129507

image

(not a blocker for this PR, but if it works, is probably a cleaner pattern for future/similar usecases)

File: src/components/templates/Jazzbar/JazzTab/JazzTab.scss
Comment (0xdevalias) [reply src/components/templates/Jazzbar/JazzTab/JazzTab.scss]: _nods_ that's fair enough. Would be good to hear your higher level thoughts on that at some point in future to get my head around your pattern/thinking a bit more.

File: src/components/templates/Jazzbar/JazzTab/JazzTab.scss
Comment (0xdevalias) [reply src/components/templates/Jazzbar/JazzTab/JazzTab.scss]: _nods_ makes sense

File: src/components/templates/Jazzbar/JazzTab/JazzTab.scss
Comment (0xdevalias) [reply src/components/templates/Jazzbar/JazzTab/JazzTab.scss]: Fair enough. Not a blocker from me. Above was suggested in line with you 'not being particularly happy with how it was implemented', and wanting to offer suggestions/alternatives that feel cleaner.

File: src/components/templates/Jazzbar/JazzTab/JazzTab.scss
Comment (0xdevalias) [reply src/components/templates/Jazzbar/JazzTab/JazzTab.scss]: Ah.. fair.. that feels.. confusing. But I guess we probably don't have a better way right now.

Perhaps adding a comment in `JazzTab.scss` on this line saying that it's used in the other file would help make this more obvious/clear in future.

File: src/components/templates/Jazzbar/JazzTab/JazzTab.scss
Comment (0xdevalias) [reply src/components/templates/Jazzbar/JazzTab/JazzTab.scss:68]: Fair enough, makes sense.

File: src/components/molecules/TableHeader/TableHeader.tsx
Comment (0xdevalias) [inline src/components/molecules/TableHeader/TableHeader.tsx:168]: Can't this be written like this?

```suggestion
              <span className="private-table-seats-left">
                {numberOfSeatsLeft} seats left
              </span>
```

File: src/components/templates/Jazzbar/JazzTab/JazzTab.scss
Comment (0xdevalias) [inline src/components/templates/Jazzbar/JazzTab/JazzTab.scss:80]: Why do we want to hide `overflow-y`? Presumably it would be better if it didn't overflow at all yeah?

File: src/components/templates/Jazzbar/JazzTab/JazzTab.scss
Comment (mike-lvov) [reply src/components/templates/Jazzbar/JazzTab/JazzTab.scss]: https://github.com/sparkletown/internal-sparkle-issues/issues/501

File: src/components/templates/Jazzbar/JazzTab/JazzTab.scss
Comment (mike-lvov) [reply src/components/templates/Jazzbar/JazzTab/JazzTab.scss]: https://github.com/sparkletown/internal-sparkle-issues/issues/501

File: src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx
Comment (mike-lvov) [reply src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx:230]: I don't think it is that easy. I guess this was my first approach. It seems like grid doesn't offer an easy solution for this. Mentioned this dicussion in https://github.com/sparkletown/internal-sparkle-issues/issues/501 though

File: src/components/templates/Jazzbar/JazzTab/JazzTab.tsx
Comment (0xdevalias) [inline src/components/templates/Jazzbar/JazzTab/JazzTab.tsx:75]: [NB] We should switch to using the react-router navigation helpers more than these explicit window.location.href things.

Is there any reason we can't at least use something from `utils/url` here?

File: src/components/templates/Jazzbar/JazzTab/JazzTab.scss
Comment (mike-lvov) [reply src/components/templates/Jazzbar/JazzTab/JazzTab.scss]: added in https://github.com/sparkletown/sparkle/pull/1217/commits/3dfa5caf56d2ba6f555b16f76265aa343b5a6771

File: src/components/templates/Jazzbar/components/JazzBarRoom/Room.scss
Comment (0xdevalias) [inline src/components/templates/Jazzbar/components/JazzBarRoom/Room.scss:87]: I don't think `&__participants` will refer to the same thing here, I think it will only 'reach' 1 level up, so you may need to use the full name for these styles.

Comment (0xdevalias) [reply src/components/templates/Jazzbar/components/JazzBarRoom/Room.scss:87]: Also, should `.jazzbar__participant` be `.jazzbar-room__participant`?

**Review by 0xdevalias — APPROVED** [review]
[CRO] Nothing big/blocking seeming here, so will give the approve (assuming it works correctly on clickthrough)

Please check my comments/questions in this review before merging though and fix any that are relevant, as there are a couple that might need fixing (but didn't feel big enough to 'request changes' over)

File: src/components/templates/Jazzbar/JazzTab/JazzTab.scss
Comment (mike-lvov) [reply src/components/templates/Jazzbar/JazzTab/JazzTab.scss:80]: Sometimes it gets a little squeezed on small-height devices. Not critical, but @Sparkle-T noticed it.

Comment (0xdevalias) [reply src/components/templates/Jazzbar/JazzTab/JazzTab.scss:80]: Should we make it scroll in that case then instead?

Comment (mike-lvov) [reply src/components/templates/Jazzbar/JazzTab/JazzTab.scss:80]: It is centered vertically. I guess it was squeezed by a few pixels, which didn't result in any visibility loss. The problem was that an ugly scrollbar appeared here. 

image

File: src/components/templates/Jazzbar/components/JazzBarRoom/Room.scss
Comment (mike-lvov) [reply src/components/templates/Jazzbar/components/JazzBarRoom/Room.scss:87]: > I don't think &__participants will refer to the same thing here, I think it will only 'reach' 1 level up, so you may need to use the full name for these styles.

I have just checked. It applies the rules correctly 

image

File: src/components/templates/Jazzbar/JazzTab/JazzTab.tsx
Comment (mike-lvov) [reply src/components/templates/Jazzbar/JazzTab/JazzTab.tsx]: changed in https://github.com/sparkletown/sparkle/pull/1217/commits/002f1b4686eae7db7a910850d16dde1407bc4d2c

File: src/components/templates/Jazzbar/JazzTab/JazzTab.scss
Comment (0xdevalias) [reply src/components/templates/Jazzbar/JazzTab/JazzTab.scss:80]: Hrmm, fair enough. Probably not a huge issue. Would probably be 'better' to figure/fix the 'few pixels', but it's unlikely to cause big issues given the current layout of it/etc. Not a blocker.

File: src/components/templates/Jazzbar/components/JazzBarRoom/Room.scss
Comment (mike-lvov) [reply src/components/templates/Jazzbar/components/JazzBarRoom/Room.scss]: > Also, should .jazzbar__participant be .jazzbar-room__participant?

changed in https://github.com/sparkletown/sparkle/pull/1217/commits/002f1b4686eae7db7a910850d16dde1407bc4d2c

File: src/components/templates/Jazzbar/JazzTab/JazzTab.tsx
Comment (0xdevalias) [inline src/components/templates/Jazzbar/JazzTab/JazzTab.tsx:216]: **Note:** This seems to be where `UserList` was removed from the JazzBar.

**Pull request merged**
